api/alembic/versions/2169aa76dfea_add_haida_gwaii_zone_to_cofc.py

```diff
@@ -0,0 +1,125 @@
+    results = res.fetchall()
+    return results[0][0]
+
+
+def increase_value_order_of_appearance(conn, fire_centre_id: int, from_index: int):
+    # due to constraint on unique value of (fire_centre_id, order_of_appearance_in_list), have to
+    # modify values of order_of_appearance_in_list one at a time, otherwise error is thrown
+    res = conn.execute(
+        sa.text('SELECT MAX(order_of_appearance_in_list) FROM planning_areas WHERE fire_centre_id = {}'.format(fire_centre_id)))
+    max_value_order = res.fetchall()[0][0]
+    index = max_value_order
+    while index > from_index:
+        conn.execute(sa.text('UPDATE planning_areas SET order_of_appearance_in_list = order_of_appearance_in_list + 1 WHERE fire_centre_id = {} AND order_of_appearance_in_list = {}'.format(fire_centre_id, index)))
+        index -= 1
+
+
+def decrease_value_order_of_appearance(conn, fire_centre_id: int, to_index: int):
+    # due to constraint on unique value of (fire_centre_id, order_of_appearance_in_list), have to
+    # modify values of order_of_appearance_in_list one at a time, otherwise error is thrown
+    res = conn.execute(
+        sa.text('SELECT MAX(order_of_appearance_in_list) FROM planning_areas WHERE fire_centre_id = {}'.format(fire_centre_id)))
+    max_value_order = res.fetchall()[0][0]
+    index = to_index
+    while index <= max_value_order:
+        conn.execute(sa.text('UPDATE planning_areas SET order_of_appearance_in_list = order_of_appearance_in_list - 1 WHERE fire_centre_id = {} AND order_of_appearance_in_list = {}'.format(fire_centre_id, index)))
+        index += 1
+
+# ------ end of helper functions ----
+
+
+def upgrade():
+    conn = op.get_bind()
+
+    cofc_id = get_fire_centre_id(conn, 'Coastal Fire Centre')
+    fraser_zone = get_zone(conn, 'Fraser Zone', cofc_id)
+    fraser_zone_id = fraser_zone[0]
+    m2_25_id = get_fuel_type_id(conn, 'M2 25%')
+
+    # 1. Increase order_of_appearance_in_list by 1 for all planning areas in COFC with order value > 1
+    increase_value_order_of_appearance(conn, cofc_id, 1)
+
+    # 2. Add Haida Gwaii planning area for COFC with order_of_appearance_in_list = 2
+    conn.execute(sa.text(
+        'INSERT INTO planning_areas (name, fire_centre_id, order_of_appearance_in_list) VALUES (\'Haida Gwaii\', {}, 2)'.format(cofc_id)))
+
+    # 3. Add Honna station (station code 93) to Haida Gwaii planning area
+    haida_gwaii_zone = get_zone(conn, 'Haida Gwaii', cofc_id)
+    haida_gwaii_zone_id = haida_gwaii_zone[0]
+    now = get_utc_now()
+    conn.execute(sa.text(
+        'INSERT INTO planning_weather_stations (station_code, fuel_type_id, planning_area_id, order_of_appearance_in_planning_area_list, create_user, create_timestamp, update_user, update_timestamp, is_deleted) VALUES (93, {}, {}, 1, \'system\', \'{}\', \'system\', \'{}\', false)'.format(m2_25_id, haida_gwaii_zone_id, now, now)))
+
+    # 4. Remove Honna station from Fraser zone in COFC
```

Comment:
I wonder if we should instead mark it as deleted and perhaps have a version number that corresponds with the requests that include it or something so that we can maintain history.

Reply:
Yeah

Reply:
The users need to keep 5-7 years of historic data in their records, but it's seen as their responsibility to store that via PDFs, not our responsibility.